modules/@apostrophecms/rich-text-widget/index.js

```diff
@@ -6,6 +6,95 @@ const cheerio = require('cheerio');
 
 module.exports = {
   extend: '@apostrophecms/widget-type',
+  cascades: [ 'linkFields' ],
+  linkFields(self, options) {
+    const linkWithType = (Array.isArray(options.linkWithType)
+      ? options.linkWithType
+      : [ options.linkWithType ]);
+
+    // Labels are not available at the time the schema is built,
+    // they are added on modulesRegistered.
+    const linkWithTypeChoices = linkWithType
+      .map(type => ({
+        label: type,
+        value: type
+      }))
+      .concat([
+        {
+          label: 'apostrophe:url',
+          value: '_url'
+        }
+      ]);
+
+    const linkWithTypeFields = linkWithType.reduce((fields, type) => {
+      const name = `_${type}`;
+      fields[name] = {
+        type: 'relationship',
+        label: type,
+        withType: type,
+        required: true,
+        max: 1,
+        browse: true,
+        if: {
+          linkTo: type
+        }
+      };
+      return fields;
+    }, {});
+    linkWithTypeFields.updateTitle = {
+      label: 'apostrophe:updateTitle',
+      type: 'boolean',
+      def: true,
+      if: {
+        $or: linkWithType.map(type => ({
+          linkTo: type
+        }))
+      }
+    };
+
+    return {
+      add: {
+        linkTo: {
+          label: 'apostrophe:linkTo',
+          type: 'select',
+          choices: linkWithTypeChoices,
+          required: true,
+          def: linkWithTypeChoices[0].value
+        },
+        ...linkWithTypeFields,
+        href: {
+          label: 'apostrophe:url',
+          type: 'string',
+          required: true,
+          if: {
+            linkTo: '_url'
+          }
+        },
+        target: {
+          label: 'apostrophe:linkTarget',
+          type: 'checkboxes',
+          htmlAttribute: true,
+          choices: [
+            {
+              label: 'apostrophe:openLinkInNewTab',
+              value: '_blank'
+            }
+          ]
+        }
+      },
+      group: {
```

Comment:
`group` doesn't really apply to a modal without tabs. It also looks like you already have the right default order here.

If you do need to order things in a schema without tabs, you can use `order` which takes an array of field names. But in this case I'm pretty sure the default behavior is already correct without either `group` or `order`.

Reply:
`order` does not re-order the schema fields but `filters/columns` etc. The only way to allow control of the field order when extending is a "fake" group as far as I know. 
Adding the defaults then allows fine tuning when extending (done in the testbed). E.g. "put target and my other single choice fields in the end because they don't have "field label" by design and look confusing when mixed with other fields", but I don't have to put the whole list of the default fields on a project level.

I'll remove the default group.